File: 2024/day-01/src/bin/part1.rs
fn main() {
    let mut lines: Vec<&str> = include_str!("../inputs/input.txt").lines().collect();
    let mut left: Vec<i32> = Vec::new();
    let mut right: Vec<i32> = Vec::new();
    for line in lines {
        let test: Vec<&str> = line.split_whitespace().collect();
        left.push(test[0].parse().unwrap());
        right.push(test[1].parse().unwrap());
    }
    left.sort();
    right.sort();
    let arr: Vec<_> = left.iter().zip(right.iter()).collect();
    let mut accumulator = 0;
    for element in arr {
        let result = element.0.abs_diff(*element.1);
        accumulator += result;
    }
    println!("{:?}", accumulator);
}

File: 2020/day-01/part1.py
with open('input.txt', 'r') as file:
    lines = [line.strip() for line in file]
    
numbers = []

for line in lines:
    numbers.append(int(line))

print(numbers)

for i in numbers:
    for j in numbers:
        if i == j:
            continue
        if i + j == 2020:
            print(i * j)
